### PR TITLE
Fixes cross platform build failure

### LINKED ISSUE #51358: cross-build: PR 49988 causing cross build failure
```
I0825 19:05:24.671] +++ [0825 19:04:12] windows/amd64: go build started
I0825 19:05:24.672] # k8s.io/kubernetes/pkg/volume/local
I0825 19:05:24.672] pkg/volume/local/local.go:306: undefined: syscall.Stat_t
```
https://github.com/kubernetes/kubernetes/pull/49988
https://k8s-testgrid.appspot.com/misc#cross-build
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-cross-build/3054

cc: @kubernetes/sig-storage-bugs @kubernetes/sig-storage-test-failures @sbezverk @msau42 @rootfs @smarterclayton

### PR DESCRIPTION
**Pull request by sbezverk**
Closes #51358

### CODE REVIEW

**Comment by msau42** [comment]
Would it be better to put this into some shared volume util library?

**Comment by sbezverk** [comment]
@msau42 if it goes into shared library, then it needs to be changed to global and also probably additional unit tests would be required.

**Comment by msau42** [comment]
Yes that sounds good :-)

File: pkg/volume/volume_windows.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
+/*
```
Comment (msau42) [inline pkg/volume/volume_windows.go]: Does this file need some flag to only build for windows?

File: pkg/volume/volume_windows.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+import (
+	"github.com/golang/glog"
+	"os"
+)
+
+// IsSameFSGroup is called only for requests to mount an already mounted
+// volume. It checks if fsGroup of new mount request is the same or not.
+// It returns false if it not the same. It also returns current Gid of a path
+// provided for dir variable.
```
Comment (msau42) [inline pkg/volume/volume_windows.go]: Add a comment that for windows we're not checking anything and just returning true

File: pkg/volume/volume_windows.go (hunk 1 above)
Comment (sbezverk) [reply pkg/volume/volume_windows.go]: I was told no flag needed for windows.

File: pkg/volume/volume_windows.go (hunk 2 above)
Comment (sbezverk) [reply pkg/volume/volume_windows.go]: +1 Good idea. Will change the comment with the next iteration.

**Comment by dims** [comment]
@sbezverk looks like you need to run `update-bazel.sh`

File: pkg/volume/volume_windows.go (hunk 1 above)
Comment (m1093782566) [reply pkg/volume/volume_windows.go]: I find CI does run windows build, see

```
I0826 07:47:28.050] +++ [0826 07:43:15] windows/amd64: go build started
I0826 07:47:28.050] +++ [0826 07:47:28] windows/amd64: go build finished
I0826 07:47:28.059] Makefile:92: recipe for target 'all' failed
I0826 07:47:28.062] Makefile:419: recipe for target 'cross' failed
```

**Comment by m1093782566** [comment]
```
I0826 07:47:28.048] +++ [0826 07:43:15] linux/amd64: go build started
I0826 07:47:28.049] +++ [0826 07:47:14] linux/amd64: go build finished
I0826 07:47:28.049] +++ [0826 07:43:15] darwin/amd64: go build started
I0826 07:47:28.049] # k8s.io/kubernetes/pkg/volume/local
I0826 07:47:28.049] pkg/volume/local/local.go:235: undefined: volume.IsSameFSGroup
I0826 07:47:28.050] !!! [0826 07:46:30] Call tree:
I0826 07:47:28.050] !!! [0826 07:46:30]  1: /go/src/k8s.io/kubernetes/hack/lib/golang.sh:696 kube::golang::build_binaries_for_platform(...)
I0826 07:47:28.050] !!! [0826 07:46:30]  2: hack/make-rules/build.sh:27 kube::golang::build_binaries(...)
I0826 07:47:28.050] +++ [0826 07:43:15] windows/amd64: go build started
I0826 07:47:28.050] +++ [0826 07:47:28] windows/amd64: go build finished
I0826 07:47:28.059] Makefile:92: recipe for target 'all' failed
I0826 07:47:28.062] Makefile:419: recipe for target 'cross' failed
```

**Comment by sbezverk** [comment]
@msau42 @cblecker all tests are green including cross platform one. please approve..

**Comment by cblecker** [comment]
/assign @rootfs
for approval

**Pull request merged**